funfact/algorithm.py

```diff
@@ -113,22 +115,23 @@ class _Factorization(Factorization, ab.AutoGradMixin):
     opt_fac = _Factorization.from_tsrex(tsrex_vec, dtype=dtype)
 
     try:
-        opt = optimizer(opt_fac.factors, lr=lr, **kwargs)
+        opt = optimizer(opt_fac.factors, lr=lr)
```

Comment:
This means that we can not set the other hyperparameters of the optimizer? E.g. `beta1`, `beta2`, `epsilon` for Adam. Maybe we can use the dictionary approach to still allow this?

Reply:
I am thinking about adapting the class-or-object idiom to optimizers too. So if the user supplies a class (or str of the class name), it will be instantiated with the default arguments. Alternatively, they can create an optimizer object on their own so that additional arguments can be given there.

Reply:
In other words, we should avoid bringing the keywords of components such as loss or optimizer into the factorize method, because:
1. It will become more and more challenging if we involve more components. For example, I am eyeing multi-tensor penalty terms for V2.0 :grin: 
2. This would make the signature more and more complicated, and also requires us to duplicate docstrings.

Reply:
Sounds like a good idea